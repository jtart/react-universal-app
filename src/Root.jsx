import React from 'react';

export const ServerData = ({ data }) => {
  const serialisedProps = JSON.stringify(data).replace(/\//g, '\\/');

  return (
    <script
      id="uni-data"
      type="application/json"
      dangerouslySetInnerHTML={{
        __html: serialisedProps,
      }}
    />
  );
};

export const Scripts = ({ scripts }) =>
  scripts.map(script => <script src={script} defer />);

export const Root = ({ html }) => (
  <div id="__uni__" dangerouslySetInnerHTML={{ __html: html }} />
);
