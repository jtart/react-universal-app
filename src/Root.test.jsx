import React from 'React';
import { ServerData, Scripts, Root } from './Root';

describe('ServerData', () => {
  it('should render a script tag containing data', () => {
    const data = { item: 'This is some data' };

    const wrapper = shallow(<ServerData data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Scripts', () => {
  it('should render scripts for each JS script', () => {
    const scripts = [
      'https://example.com/bundle.one.js',
      'https://example.com/bundle.two.js',
    ];

    const wrapper = shallow(<Scripts scripts={scripts} />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Root', () => {
  it('should render a div with injected HTML', () => {
    const html = <div>Hello!</div>;

    const wrapper = shallow(<Root html={html} />);
    expect(wrapper).toMatchSnapshot();
  });
});
