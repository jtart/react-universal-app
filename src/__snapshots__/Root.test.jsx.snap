// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should render a div with injected HTML 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": <div>
        Hello!
      </div>,
    }
  }
  id="__uni__"
/>
`;

exports[`Scripts should render scripts for each JS script 1`] = `
Array [
  <script
    defer={true}
    key="script-https://example.com/bundle.one.js"
    src="https://example.com/bundle.one.js"
  />,
  <script
    defer={true}
    key="script-https://example.com/bundle.two.js"
    src="https://example.com/bundle.two.js"
  />,
]
`;
